src/app/api/upload/route.ts

```diff
@@ -0,0 +1,56 @@
+import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { NextRequest, NextResponse } from 'next/server';
+import { v4 as uuidv4 } from 'uuid';
+
+const AWS_S3_ACCESS_ID = process.env.AWS_S3_ACCESS_ID || '';
+const AWS_S3_ACCESS_KEY = process.env.AWS_S3_ACCESS_KEY || '';
+const AWS_S3_REGION = process.env.AWS_S3_REGION || '';
+const AWS_S3_BUCKET = process.env.AWS_S3_BUCKET || '';
+
+const client = new S3Client({
+  region: AWS_S3_REGION,
+  credentials: {
+    accessKeyId: AWS_S3_ACCESS_ID,
+    secretAccessKey: AWS_S3_ACCESS_KEY,
+  },
+});
+
+// 이미지를 S3에 업로드하는 함수
+export const POST = async (request: NextRequest): Promise<string | unknown> => {
+  const formData = await request.formData();
+  const files = formData.getAll('files') as File[];
+
+  const uploadPromises = files.map(async (file) => {
+    const arrayBuffer = await file.arrayBuffer();
+    const buffer = Buffer.from(arrayBuffer);
+    const key = `${uuidv4()}/${file.name}`;
+
+    const command = new PutObjectCommand({
+      Bucket: AWS_S3_BUCKET,
+      Key: key,
+      Body: buffer,
+      ContentType: file.type,
+    });
+
+    try {
+      const response = await client.send(command);
+      return response.$metadata.httpStatusCode === 200
+        ? `https://${AWS_S3_BUCKET}.s3.${AWS_S3_REGION}.amazonaws.com/${key}`
+        : '';
+    } catch (error) {
+      console.error('Error uploading file:', error);
+      throw error;
```

Comment:
밑에서는 처리하신 것 같은데 이쪽 부분에서도 에러가 발생했다면 에러 응답 코드를 가진 `NextResponse`를 보내면 좋을것 같습니다
```suggestion
return NextResponse.json(
      { message: 'Failed to upload images' },
      { status: 500 },
    );
```

Reply:
이 부분은 의도적으로 error를 throw 하도록 한 거에요! 잘 보시면 이미지 1장을 업로드하는 callback을 map을 통해서 `uploadPromises`에 저장하고 `Promise.all`을 통해 동시에 업로드하는 과정이에요. 이 때, 에러가 발생할 경우 throw해서 아래의 try catch문에서 catch해서 500에러 응답을 보내도록 했어요. 따라서 해당 과정에서 에러가 발생할 경우 NextResponsefh 500 에러를 발생시키기 때문에 문제가 없습니다!

Reply:
아하 `Promise.all`을 못보고 지나쳤네요 all을 하면 하나라도 프라미스가 정상작동되지 않으면 에러를 리턴하는 특징을 까먹고 있었는데 덕분에 다시 상기하고 갑니다!